src/main/java/io/xdb/core/communication/CommunicationSchema.java

```diff
@@ -0,0 +1,12 @@
+package io.xdb.core.communication;
+
+import java.util.List;
+import lombok.Builder;
+import lombok.Getter;
+
+@Builder
+@Getter
+public class CommunicationSchema {
```

Comment:
😨 I think you misunderstood what I meant (sorry i didn't make it clear)

I think NOT having this communicationSchema is better because it saves the code for user, and server doesn't care about it at all(unlike local/global attributes, server need to know what to load or lock)

Reply:
Then how to decode the payload with an objectEncoder and a class type? Where are you going to store such info?

Reply:
Or I think we need to pass an objectEncoder to the execute method.

Reply:
we can require a type class to be passed like this: https://github.com/indeedeng/iwf-java-sdk/blob/b87c3883d664b3c08deeb3f9a8e472b896e304a3/src/main/java/io/iworkflow/core/Client.java#L254